File: _assets/output_gen/Entities/Candidate.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace OB.Data.Entities
{
    public class Candidate
    {
        //--------------------------------------------------------
        [Key]
        public int Id { get; set;}

        public int OfficeId { get; set;}

        [Required,StringLength(32)]
        public string FirstName { get; set;}

        [StringLength(32)]
        public string MiddleName { get; set;}

        [Required,StringLength(32)]
        public string LastName { get; set;}

        [Required,StringLength(32)]
        public string Party { get; set;}

        public bool IsSelected { get; set;}

        public bool IsIncumbant { get; set;}

        [Required]
        public DateTime DateCreated { get; set;}

        public DateTime DateUpdated { get; set;}

        public string UpdatedBy { get; set;}

        //--------------------------------------------------------
    }
}

File: _assets/output_gen/jsModels/Candidate.js

//---------------------------------------------------------------------------

export default class Candidate {
  constructor(options) {

    this.Id = null;
    this.OfficeId = null;
    this.FirstName = null;
    this.MiddleName = null;
    this.LastName = null;
    this.Party = null;
    this.IsSelected = null;
    this.IsIncumbant = null;
    this.DateCreated = null;
    this.DateUpdated = null;
    this.UpdatedBy = null;

    if(options) {
      this.init(options);
    }
  }

  //---------------------------------------------------------------------------
  init(options) {

    this.Id = options.Id || this.Id;
    this.OfficeId = options.OfficeId || this.OfficeId;
    this.FirstName = options.FirstName || this.FirstName;
    this.MiddleName = options.MiddleName || this.MiddleName;
    this.LastName = options.LastName || this.LastName;
    this.Party = options.Party || this.Party;
    this.IsSelected = options.IsSelected || this.IsSelected;
    this.IsIncumbant = options.IsIncumbant || this.IsIncumbant;
    this.DateCreated = options.DateCreated || this.DateCreated;
    this.DateUpdated = options.DateUpdated || this.DateUpdated;
    this.UpdatedBy = options.UpdatedBy || this.UpdatedBy;

  }  
  //---------------------------------------------------------------------------  
}